File: withJava/src/main/java/id/bakode/mvp/java/MainActivity.java
package id.bakode.mvp.java;

import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity implements MainView {

    private MainPresenter mPresenter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        mPresenter = new MainPresenter(this, new MainInteractor());
        doJob();
    }

    private void doJob() {
        mPresenter.calculate();
    }
}

File: withKotlin/src/main/java/id/bakode/mvp/kotlin/MainActivity.kt
package id.bakode.mvp.kotlin

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle


class MainActivity : AppCompatActivity(), MainView {

    private lateinit var mPresenter: MainPresenter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        mPresenter = MainPresenter(this, MainInteractor())

        doJob()
    }

    private fun doJob() {
        mPresenter.calculate()
    }
}
